great_expectations/experimental/datasources/data_asset/data_connector/file_path_data_connector.py

```diff
@@ -0,0 +1,303 @@
+batch identifiers {batch_definition.batch_identifiers} from batch definition {batch_definition}.
+"""
+            )
+
+        path = self._get_full_file_path(path=path)
+
+        return {"path": path}
+
+    # Interface Method
+    def _refresh_data_references_cache(self) -> None:
+        # Map data_references to batch_definitions
+        self._data_references_cache = {}
+
+        for data_reference in self._get_data_reference_list():
+            mapped_batch_definition_list: Optional[
+                List[BatchDefinition]
+            ] = self._map_data_reference_to_batch_definition_list(
+                data_reference=data_reference
+            )
+            self._data_references_cache[data_reference] = mapped_batch_definition_list
+
+    # Interface Method
+    def _map_data_reference_to_batch_definition_list(
+        self, data_reference: str
+    ) -> Optional[List[BatchDefinition]]:
+        return map_data_reference_string_to_batch_definition_list_using_regex(
+            datasource_name=self.datasource_name,
+            data_connector_name=self.name,
+            data_asset_name=self.data_asset_name,
+            data_reference=data_reference,
+            regex_pattern=self._regex,
+        )
+
+    # Interface Method
+    def _map_batch_definition_to_data_reference(
+        self, batch_definition: BatchDefinition
+    ) -> str:
+        group_names: List[str] = self._regex_parser.get_all_group_names()
+        return map_batch_definition_to_data_reference_string_using_regex(
+            batch_definition=batch_definition,
+            regex_pattern=self._regex,
+            group_names=group_names,
+        )
+
+    # TODO: <Alex>ALEX_INCLUDE_SORTERS_FUNCTIONALITY_UNDER_PYDANTIC-MAKE_SURE_SORTER_CONFIGURATIONS_ARE_VALIDATED</Alex>
+    # TODO: <Alex>ALEX</Alex>
+    # def _sort_batch_definition_list(
+    #     self, batch_definition_list: List[BatchDefinition]
+    # ) -> List[BatchDefinition]:
+    #     """
+    #     Use configured sorters to sort batch_definition
+    #
+    #     Args:
+    #         batch_definition_list (list): list of batch_definitions to sort
+    #
+    #     Returns:
+    #         sorted list of batch_definitions
+    #
+    #     """
+    #     sorters: Iterator[Sorter] = reversed(list(self.sorters.values()))
+    #     for sorter in sorters:
+    #         batch_definition_list = sorter.get_sorted_batch_definitions(
+    #             batch_definitions=batch_definition_list
+    #         )
+    #
+    #     return batch_definition_list
+    # TODO: <Alex>ALEX</Alex>
+
+    def _get_batch_definition_list_from_cache(self) -> List[BatchDefinition]:
+        batch_definition_list: List[BatchDefinition] = [
+            batch_definitions[0]
+            for batch_definitions in self._data_references_cache.values()
+            if batch_definitions is not None
+        ]
+        return batch_definition_list
+
+    # Interface Method
+    def _get_data_reference_list(self) -> List[str]:
```

Comment:
why do we have to redefine this here as (again) not implemented?

Reply:
@donaldheppner This is not a redefinition -- please note that the module you are referring to is now in ZEP (not V3), and the interface is different -- in V3, this method also accepts `data_asset_name` (this has no meaning in ZEP, because `DataAsset` is the class that uses `DataConnector`, not the other way around -- as is in V3).

Reply:
@donaldheppner I did some research, and found that at least according to "https://stackoverflow.com/questions/34066933/pycharm-warning-must-implement-all-abstract-methods#:~:text=We%20can%20disable%20this%20warning,uncheck%20it%2C%20then%20click%20apply", the recommendation is to disable the `PyCharm` warning, because it is considered a weak warning (in fact, an issue was filed with `PyCharm` to handle this more gracefully, but it has not been resolved as of yet).  In any case, the duplicated interface method declarations have been deleted.  Thank you.

Reply:
then why is it still here?

Reply:
@donaldheppner Oops -- sorry -- good catch; I deleted the repetition in ZEP -- now it is gone here as well.  Thanks.